metricbeat/mb/event.go

```diff
@@ -141,6 +150,9 @@ func AddMetricSetInfo(module, metricset string, event *Event) {
 	if event.Took > 0 {
 		e.Put("event.duration", event.Took/time.Nanosecond)
 	}
+	if event.Period > 0 {
+		e.Put("metricset.period", event.Period/time.Microsecond)
```

Comment:
I understand why you want here with microseconds instead of nano seconds as it might be overkill. At the same time I wonder if we should standardise the unit we use?

Otherwise, do we expected / recommend anyhting lower then 1s?

Reply:
> I understand why you want here with microseconds instead of nano seconds as it might be overkill. At the same time I wonder if we should standardise the unit we use?

I don't have a strong opinion about this, it comes from the original PR :slightly_smiling_face:
By standardizing do you mean to use nanoseconds as in `event.duration`? I would be ok with that, but I don't think they need to be in the same unit as they have different needings.

> Otherwise, do we expected / recommend anyhting lower then 1s?

I don't think it will be recommendable to have a period lower than the second, but there are services that can very well respond in milliseconds, so it should be possible.

Another option can be to store it in seconds, but as a float, this way we would need less space in events and in store (32 bits for a float, 64 for a long), and we could still support the case of periods under the second.

Reply:
I definitively prefer the long over the float as the float gets inaccurate in ES for large values (not that we will really hit those hopefully). I'm still holding my breath for getting duration support in ES one day ;-)

I'm good with Micro or either Milli and we solve it in case it ever becomes an issue.

Reply:
Ok, I think I am going to change this to Milliseconds, it should be more than enough for this. And then also integer should be enough to store the values.